config_modules/list_to_map/main.tf

```diff
@@ -0,0 +1,13 @@
+##############################################################################
+# Fail States
+##############################################################################
+
+locals {
+  configuration_failure_conflicting_values_lookup_value_regex_and_value_is_not_null = regex("false", (
+    var.value_is_not_null == true && var.lookup_value_regex != null
+    ? true
+    : false
+  ))
+}
+
+##############################################################################
```

Comment:
Are we repeating code accross module. for example, I see we use a list_to_map module somewhere else too -> https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/blob/main/dynamic_values/config_modules/list_to_map/list_to_map.tf

Have you considered creating an external module for this so we can keep module code more DRY?